irksome/explicit_stepper.py

```diff
@@ -20,3 +20,16 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
             F, butcher_tableau, t, dt, u0, bcs=bcs,
             solver_parameters=solver_parameters, appctx=appctx,
             nullspace=None)
+
+    # DAE treatment of the boundary conditions in this case says that
+    # we should impose the BCs so that they are satisfied for the next
+    # stage for all but that last stage, and that they are satisfied
+    # for the next timestep for the last stage.
+    def update_bc_constants(self, AAb, CCone, i, a_vals, d_val, c):
+        ns = AAb.shape[1]
+        for j in range(i):
+            a_vals[j].assign(AAb[i+1, j])
+        for j in range(i, ns):
+            a_vals[j].assign(0)
+        d_val.assign(AAb[i+1, i])
+        c.assign(CCone[i+1])
```

Comment:
Could this shft be avoided by truncating the first row of `AAb`?

Reply:
I was trying to write this in a way that didn't need any logic to handle the DIRK vs. explicit cases outside of how we update the constants in the BCs.  I guess we could exchange this for having different constructions of `AAb` and `CCone` in the `__init__` class, but I'm not sure we gain anything by it...